crates/bevy_pbr/src/light_probe/mod.rs

```diff
@@ -119,24 +172,123 @@ pub struct LightProbesBuffer(DynamicUniformBuffer<LightProbesUniform>);
 pub struct ViewLightProbesUniformOffset(u32);
 
 /// Information that [`gather_light_probes`] keeps about each light probe.
-#[derive(Clone, Copy)]
+///
+/// This information is parameterized by the [`LightProbeComponent`] type. This
+/// will either be [`EnvironmentMapLight`] for reflection probes or
+/// [`IrradianceVolume`] for irradiance volumes.
 #[allow(dead_code)]
-struct LightProbeInfo {
+struct LightProbeInfo<C>
+where
+    C: LightProbeComponent,
+{
     // The transform from world space to light probe space.
     inverse_transform: Mat4,
 
     // The transform from light probe space to world space.
     affine_transform: Affine3A,
 
-    // The diffuse and specular environment maps associated with this light
-    // probe.
-    environment_maps: EnvironmentMapIds,
-
     // Scale factor applied to the diffuse and specular light generated by this
     // reflection probe.
     //
     // See the comment in [`EnvironmentMapLight`] for details.
     intensity: f32,
+
+    // The IDs of all assets associated with this light probe.
+    //
+    // Because each type of light probe component may reference different types
+    // of assets (e.g. a reflection probe references two cubemap assets while an
+    // irradiance volume references a single 3D texture asset), this is generic.
+    asset_id: C::AssetId,
+}
+
+/// A component, part of the render world, that stores the mapping from asset ID
+/// or IDs to the texture index in the appropriate binding arrays.
+///
+/// Cubemap textures belonging to environment maps are collected into binding
+/// arrays, and the index of each texture is presented to the shader for runtime
+/// lookup. 3D textures belonging to reflection probes are likewise collected
+/// into binding arrays, and the shader accesses the 3D texture by index.
+///
+/// This component is attached to each view in the render world, because each
+/// view may have a different set of light probes that it considers and therefore
+/// the texture indices are per-view.
+#[derive(Component, Default)]
+pub struct RenderViewLightProbes<C>
+where
+    C: LightProbeComponent,
+{
+    /// The list of environment maps presented to the shader, in order.
+    binding_index_to_textures: Vec<C::AssetId>,
+
+    /// The reverse of `binding_index_to_cubemap`: a map from the texture ID to
+    /// the index in `binding_index_to_cubemap`.
+    cubemap_to_binding_index: HashMap<C::AssetId, u32>,
+
+    /// Information about each light probe, ready for upload to the GPU, sorted
+    /// in order from closest to the camera to farthest.
```

Comment:
Shouldnt this be sorted by screen coverage? If a light probe is really close to the camera but behind it, its not useful. If a light probe is really large and covers a lot of the screen but its center is far from the camera, it should still be more/as important as small lightprobes that are closer by.

This can be handled in a followup.

Reply:
Yes, eventually what we want to do is to use clustering just like we use for lights. This logic is a temporary hack for now to reduce the complexity of this patch.